pkg/cluster/defaults.go

```diff
@@ -28,3 +31,42 @@ func (d ControlPlaneIPCheckAnnotationDefaulter) ControlPlaneIPCheckDefault(ctx c
 
 	return spec, nil
 }
+
+// MachineHealthCheckDefaulter is the defaulter created to configure the machine health check timeouts.
+type MachineHealthCheckDefaulter struct {
+	nodeStartupTimeout      string
+	unhealthyMachineTimeout string
+}
+
+// NewMachineHealthCheckDefaulter allows to create a new MachineHealthCheckDefaulter.
+func NewMachineHealthCheckDefaulter(nodeStartupTimeout, unhealthyMachineTimeout string) MachineHealthCheckDefaulter {
+	return MachineHealthCheckDefaulter{
+		nodeStartupTimeout:      nodeStartupTimeout,
+		unhealthyMachineTimeout: unhealthyMachineTimeout,
+	}
+}
+
+// MachineHealthCheckDefault sets the defaults for machine health check timeouts.
+func (d MachineHealthCheckDefaulter) MachineHealthCheckDefault(ctx context.Context, spec *Spec) (*Spec, error) {
```

Comment:
We don't need an error here then

Reply:
here as well since the defaulter needs the signature that returns an error as well as the spec, I made it compliant.